File: server/src/main/java/de/be/bpv/server/rest/export/request/ExportRequest.java
package de.be.bpv.server.rest.export.request;

import com.fasterxml.jackson.annotation.JsonInclude;

/**
 * Request for exporting a CAD file and mapping.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ExportRequest {

    /**
     * ID of the CAD file to export.
     */
    private Long cadFileId;

    /**
     * ID of the mapping to export.
     */
    private Long mappingId;

    /**
     * Name of the color map to use.
     */
    private String colorMap;

    public Long getCadFileId() {
        return cadFileId;
    }

    public void setCadFileId(Long cadFileId) {
        this.cadFileId = cadFileId;
    }

    public Long getMappingId() {
        return mappingId;
    }

    public void setMappingId(Long mappingId) {
        this.mappingId = mappingId;
    }

    public String getColorMap() {
        return colorMap;
    }

    public void setColorMap(String colorMap) {
        this.colorMap = colorMap;
    }

}

File: client/src/app/service/export/export-request.ts
/**
 * Export request.
 */
export interface ExportRequest {

	/**
	 * ID of the CAD file to export.
	 */
	cadFileId: number;

	/**
	 * ID of the room mapping collection to export.
	 */
	mappingId?: number;

	/**
	 * Name of the color map to export with.
	 */
	colorMap?: string;

}
